share/spice/whois/whois.js

```diff
@@ -66,8 +73,10 @@
         }
 
         // check we have 2 nameserver hostnames
-        if(nsObj && nsObj.hostNames[0] && nsObj.hostNames[1]) {
-            var nameServers = [nsObj.hostNames[0].toLowerCase(),nsObj.hostNames[1].toLowerCase()].join(' ');
+        if(nsObj.hostNames) {
+            if(nsObj.hostNames.length >= 2) {
+                var nameServers = [nsObj.hostNames[0].toLowerCase(),nsObj.hostNames[1].toLowerCase()].join(' ');
```

Comment:
@MrChrisW FWIW when grabbing deep elements of objects that may be undefined it might help to use our [DDG.getProperty](https://duck.co/duckduckhack/spice_js_api#getpropertyobj-pathname) function. It will catch the undefined and return null